xclim/core/calendar.py

```diff
@@ -834,14 +914,99 @@ def construct_offset(mult: int, base: str, start_anchored: bool, anchor: str | N
     -----
     This provides the mirror opposite functionality of :py:func:`parse_offset`.
     """
-    start = "S" if start_anchored and base in "YAQM" else ""
+    version = _get_freq_version(version)
+    if version == "new":
+        defend = "E"
+    else:
+        defend = ""
+    if base in "YAQM":
+        start = "S" if start_anchored else defend
+    else:
+        start = ""
     if anchor is None and base in "AQY":
         anchor = "JAN" if start_anchored else "DEC"
+    if version in ["inter", "old"] and base == "Y":
+        if anchor == "JAN":
+            anchor = ""
+        else:
+            base = "A"
     return (
         f"{mult if mult > 1 else ''}{base}{start}{'-' if anchor else ''}{anchor or ''}"
     )
 
 
+FREQ_OLD_NEW = {
+    "H": "h",
+    "T": "min",
+    "S": "s",
+    "L": "ms",
+    "U": "us",
+    "N": "ns",
+}
+
+
+# TODO: This can be removed when we pin pandas >= 2.2
```

Comment:
Maybe we can integrate a small helper function here to check what the required pins are?

Something like:
```
from importlib.metadata import distribution

for req in distribution("xclim").requires:
    if req.split(";")[0].startswith("pandas>=2.2"):
        raise FixMeOrSomethingPleaseException()
```